src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.OSX/CertificatePal.cs

```diff
@@ -381,6 +382,12 @@ public DSA GetDSAPrivateKey()
             SafeSecKeyRefHandle publicKey = Interop.AppleCrypto.X509GetPublicKey(_certHandle);
             SafeSecKeyRefHandle privateKey = Interop.AppleCrypto.X509GetPrivateKeyFromIdentity(_identityHandle);
 
+            if (publicKey.IsInvalid)
+            {
```

Comment:
Should `publicKey` be Dispose'd here to prevent the unnecessary finalization costs?